### PR TITLE
Added Spotless plugin to prettify Java code(as pylint) and Refactored Tests

### PR DESCRIPTION
**Pull request by ashutoshcp**
* Added IntegrationTestSuite [One place for running all tests]
* Refactored tests [moved repeating Lines of codes in test to BaseTest.java]
* In  OpenLocationCode, replaced deprecated usage of members.
* Added a few lines to .gitignore, IntelliJ Idea Related files exclusion.

### CODE REVIEW

**Comment by drinckes** [comment]
The Bazel tests are not passing - could you take a look? You probably need to add the BaseTest.java file into the BUILD rules.

Also - be aware that there is a rounding bug in nearly all implementations (see issue #307). There are a few implementations that have a fix for this, you could take a look at the go or js code.

**Comment by ashutoshcp** [comment]
@drinckes can you check this now. No issue in Java tests. It's passing successfully.

File: java/src/test/java/com/google/openlocationcode/EncodingTest.java [hunk 1]
```diff
@@ -57,30 +55,23 @@ public void setUp() throws Exception {
...
-              testData.latitude,
-              testData.longitude,
-              testData.length),
+          "Latitude :"
+              + latitude
+              + ", longitude:"
+              + longitude
+              + " and length:"
+              + length
+              + " were wrongly encoded.",
```
Comment (zongweil) [inline java/src/test/java/com/google/openlocationcode/EncodingTest.java]: Why the change? This looks less readable to me.

File: java/BUILD [hunk 2]
```diff
@@ -6,96 +6,23 @@ java_library(
 )
 
 java_test(
-    name = "DecodingTest",
```
Comment (zongweil) [inline java/BUILD]: What's the motivation behind this change?

File: java/src/test/java/com/google/openlocationcode/EncodingTest.java (hunk 1 above)
Comment (ashutoshcp) [reply java/src/test/java/com/google/openlocationcode/EncodingTest.java]: It's cause Used Google Java code formatting(1.7). 
Added spotless that will apply on java files and formats the code.

File: java/BUILD (hunk 2 above)
Comment (ashutoshcp) [reply java/BUILD]: By mistake, no motivation, needs to be changed

Comment (ashutoshcp) [reply java/BUILD]: Will Rename

File: java/src/test/java/com/google/openlocationcode/BaseTest.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.google.openlocationcode;
+
+import java.io.File;
+
+/**
+ * Test class for accumulating all the general methods at one place this will provide more
+ * re-usability. KLOC in the tests will be decreased.
+ */
+public class BaseTest {
```
[the remark this answers is not in the document]
Comment (ashutoshcp) [reply java/src/test/java/com/google/openlocationcode/BaseTest.java]: Yes make sense, appreciate your effort, I'll update this.

File: test_data/encoding_java.csv [hunk 4]
```diff
@@ -0,0 +1,115 @@
+# Test encoding Open Location Codes.
```
[the remark this answers is not in the document]
Comment (ashutoshcp) [reply test_data/encoding_java.csv]: Few of tests were failing, I'll remove this file and redirect to original one only.

File: java/BUILD (hunk 2 above)
Comment (ashutoshcp) [reply java/BUILD]: you will have all the test at a single place.
you need to add in BUILD=> srcs and IntegrationTestSuite.
No need to add explict function for each test.

Comment (ashutoshcp) [reply java/BUILD]: See the content of BUILD file, now it becomes so less and easy to change.

File: test_data/encoding_java.csv (hunk 4 above)
[the remark this answers is not in the document]
Comment (ashutoshcp) [reply test_data/encoding_java.csv]: Removed this file

File: java/src/test/java/com/google/openlocationcode/BaseTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (ashutoshcp) [reply java/src/test/java/com/google/openlocationcode/BaseTest.java]: updated

File: java/BUILD (hunk 2 above)
Comment (ashutoshcp) [reply java/BUILD]: See the content of BUILD

File: java/pom.xml [hunk 5]
```diff
@@ -87,6 +87,31 @@
           <autoReleaseAfterClose>true</autoReleaseAfterClose>
         </configuration>
       </plugin>
+      <plugin>
```
Comment (ashutoshcp) [inline java/pom.xml]: Formatting source code using this Spotless plugin.
for https://github.com/google/open-location-code/pull/325/files#r281754908
You just need to run the spotless apply from plugins to format the source code. 
No human effort involved anymore.

File: java/src/test/java/com/google/openlocationcode/EncodingTest.java (hunk 1 above)
Comment (ashutoshcp) [reply java/src/test/java/com/google/openlocationcode/EncodingTest.java]: https://github.com/google/open-location-code/pull/325/files#r282155179

Comment (ashutoshcp) [reply java/src/test/java/com/google/openlocationcode/EncodingTest.java]: Spotless is  to Java, like pylint to python.

**Comment by ashutoshcp** [comment]
Hi @zongweil  can you please review this..

**Comment by zongweil** [comment]
I'm traveling right now, will try to review tomorrow.

File: java/src/test/java/com/google/openlocationcode/EncodingTest.java (hunk 1 above)
Comment (drinckes) [reply java/src/test/java/com/google/openlocationcode/EncodingTest.java]: I agree with @zongweil  - that looks horrible. If you need a formatter please use https://github.com/google/google-java-format (which is what the original code was based on).

Comment (ashutoshcp) [reply java/src/test/java/com/google/openlocationcode/EncodingTest.java]: @drinckes  i have used the same. 
Spotless plugin with https://github.com/google/google-java-format 1.7

Comment (ashutoshcp) [reply java/src/test/java/com/google/openlocationcode/EncodingTest.java]: See this plugin:  https://github.com/google/open-location-code/blob/5dfd58e443f6651f0ff61db991dc8a1fd3ad3b18/java/pom.xml#L90

this creates a plugin profile in IntelliJ, with the help of that you can easily apply the formatting. that is using the same google java formatter 1.7

Comment (ashutoshcp) [reply java/src/test/java/com/google/openlocationcode/EncodingTest.java]: @drinckes @zongweil, I do realize my mistake and corrected. thanks for the review and I do apologize for my mistake.

Comment (ashutoshcp) [reply java/src/test/java/com/google/openlocationcode/EncodingTest.java]: I have updated this, hope it's fine.

File: java/BUILD (hunk 2 above)
Comment (zongweil) [reply java/BUILD]: Thanks for the context. It's actually better to have a single target per test class as opposed to a single test target with all the test classes. Bazel will manage sharding for the tests more efficiently. Could you please revert these changes?

In addition, this isn't really an integration test. A collection of unit tests is different from an integration test.

File: java/src/test/java/com/google/openlocationcode/TestUtils.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package com.google.openlocationcode;
+
+import java.io.File;
+
+/**
```
Comment (zongweil) [inline java/src/test/java/com/google/openlocationcode/TestUtils.java]: You can remove the javadocs here. They don't provide any useful info, as the class is pretty self-explanatory.

File: java/src/test/java/com/google/openlocationcode/EncodingTest.java [hunk 7]
```diff
@@ -46,7 +44,7 @@ public TestData(String line) {
 
   @Before
   public void setUp() throws Exception {
-    InputStream testDataStream = new FileInputStream(getTestFile());
+    InputStream testDataStream = new FileInputStream(TestUtils.getTestFile("encoding.csv"));
```
Comment (zongweil) [inline java/src/test/java/com/google/openlocationcode/EncodingTest.java]: The spacing looks off here. Can you re-run spotless apply?

File: java/BUILD (hunk 2 above)
Comment (ashutoshcp) [reply java/BUILD]: Reverted

File: java/src/test/java/com/google/openlocationcode/TestUtils.java (hunk 6 above)
Comment (ashutoshcp) [reply java/src/test/java/com/google/openlocationcode/TestUtils.java]: removed

File: java/src/test/java/com/google/openlocationcode/EncodingTest.java (hunk 7 above)
Comment (ashutoshcp) [reply java/src/test/java/com/google/openlocationcode/EncodingTest.java]: This seems fine. After re-run there is no change.

**Comment by ashutoshcp** [comment]
@zongweil  Renamed PR and updated with new changes.

File: java/src/test/java/com/google/openlocationcode/EncodingTest.java (hunk 7 above)
Comment (zongweil) [reply java/src/test/java/com/google/openlocationcode/EncodingTest.java]: That's fine, I'll fix it manually later.
